Applications/Spire/Include/Spire/Ui/DropDownMenu2.hpp

```diff
@@ -0,0 +1,132 @@
+#ifndef SPIRE_DROP_DOWN_MENU_2_HPP
+#define SPIRE_DROP_DOWN_MENU_2_HPP
+#include <Beam/SignalHandling/ConnectionGroup.hpp>
+#include <QVBoxLayout>
+#include "Spire/Spire/Spire.hpp"
+#include "Spire/Ui/CustomQtVariants.hpp"
+#include "Spire/Ui/DropDownItem.hpp"
+#include "Spire/Ui/DropDownWindow.hpp"
+#include "Spire/Ui/ScrollArea.hpp"
+
+namespace Spire {
+
+  //! Represents a Spire-styled drop down menu.
+  class DropDownMenu2 : public DropDownWindow {
```

Comment:
Why does this inherit from ```DropDownWindow```?

Reply:
It simplifies the initialization of the DropDownMenu and DropDownWindow by treating them as one. I could make the inputs responsible for creating and initializing the DropDownMenu within the DropDownWindow, kind of like we do for the DropShadow.

Reply:
Inheritance is never used for implementation purposes under any scenario. In the long run inheritance never simplifies anything, it only makes working with the codebase more painful.

Reply:
It may be better to revisit the DropDownWindow and address this inheritance issue then, because changing the DropDownMenu's base class will require changes to the DropDownWindow for this specific case. Changing the responsibilities from child to parent breaks some of the assumptions about how the DropDownWindow is supposed to behave.

It may be easier to have the DropDownMenu manage itself as a window instead of using the DropDownWindow, which could specifically be for pop-up windows like the color selector with multiple child widgets.

Reply:
Where else is DropDownWindow used?

Reply:
I don't see any use of DropDownWindow. Remove it from the inheritance hierarchy.